rpc_util.go

```diff
@@ -755,15 +760,19 @@ func decompress(compressor encoding.Compressor, d []byte, maxReceiveMessageSize
 // dc takes precedence over compressor.
 // TODO(dfawley): wrap the old compressor/decompressor using the new API?
 func recv(p *parser, c baseCodec, s *transport.Stream, dc Decompressor, m interface{}, maxReceiveMessageSize int, payInfo *payloadInfo, compressor encoding.Compressor) error {
-	d, err := recvAndDecompress(p, s, dc, maxReceiveMessageSize, payInfo, compressor)
+	buf, err := recvAndDecompress(p, s, dc, maxReceiveMessageSize, payInfo, compressor)
 	if err != nil {
 		return err
 	}
-	if err := c.Unmarshal(d, m); err != nil {
+	if err := c.Unmarshal(buf, m); err != nil {
 		return status.Errorf(codes.Internal, "grpc: failed to unmarshal the received message: %v", err)
 	}
-	if payInfo != nil {
-		payInfo.uncompressedBytes = d
+	if payInfo != nil && len(buf) != 0 {
+		payInfo.uncompressedBytes = make([]byte, len(buf))
+		copy(payInfo.uncompressedBytes, buf)
```

Comment:
Can we make this copy only when `p.useBytesPool` is set? That way, if the dial option is not set, we don't introduce a copy that wasn't around earlier.

Reply:
e7462501868ba7985b6e62593448e6ef5362be63 solved it.
But we don't need to take a closer look because 711db78346a7cbee693dd11f7751d64c9ab49600 is now in effect.